icubam/messaging/telegram/updater.py

```diff
@@ -0,0 +1,101 @@
+from absl import logging
+from typing import Dict
+
+from icubam.messaging.telegram import bot
+from icubam.www import token
+
+
+class UpdateFetcher:
+  """Constantly fetches updates from the telegram server.
+  
+  The UpdateFetcher is only used in dev mode when the webhook is not
+  for convenience.
+  """
+  def __init__(self, config, queue, tg_bot=None):
+    self.config = config
+    self.queue = queue
+    self.bot = bot.TelegramBot(config) if tg_bot is None else tg_bot
+    self.last_update_id = 0
+
+  async def fetch(self):
+    """Fetches updates from telegram and put it to the update queue."""
+    updates = await self.bot.getUpdates(min_id=self.last_update_id)
+    if not updates:
+      return
+
+    last_id = updates[-1]['update_id']
+    if last_id > self.last_update_id:
+      self.last_update_id = last_id
+
+    for update in updates:
+      await self.queue.put(update)
+
+
+class UpdateProcessor:
+  """What to do when receiving an update from telegram ?
+
+  As of today, if it is the first time we see this user, and it comes with
+  the proper token, then we register it and update the scheduler about it.
+  Otherwise we simply do nothing.
+  """
+  def __init__(self, config, db, queue, scheduler, tg_bot=None):
+    self.config = config
+    self.db = db
+    self.queue = queue
+    self.scheduler = scheduler
+    self.bot = bot.TelegramBot(config) if tg_bot is None else tg_bot
+    # TODO(olivier): replace with authenticator.
+    self.token_encoder = token.TokenEncoder(self.config)
+    admins = self.db.get_admins()
+    if admins:
+      self.admin_id = admins[0].user_id
+    else:
+      self.admin_id = self.db.add_default_admin()
```

Comment:
Could you please add an inline comment on why we need these admin_id when updating users?

Reply:
sure. though this is how the store is designed.